pennylane_lightning/core/src/simulators/lightning_qubit/bindings/LQubitBindings.hpp

```diff
@@ -162,12 +162,68 @@ void registerControlledGate(PyClass &pyclass) {
  */
 template <class StateVectorT, class PyClass>
 void registerBackendClassSpecificBindings(PyClass &pyclass) {
+    using PrecisionT =
+        typename StateVectorT::PrecisionT; // Statevector's precision
+    using ComplexT = typename StateVectorT::ComplexT;
+    using ParamT = PrecisionT; // Parameter's data precision
+    using np_arr_c = py::array_t<std::complex<ParamT>,
+                                 py::array::c_style | py::array::forcecast>;
+
     registerGatesForStateVector<StateVectorT>(pyclass);
     registerControlledGate<StateVectorT>(pyclass);
-    pyclass.def("applyControlledMatrix", &applyControlledMatrix<StateVectorT>,
-                "Apply controlled operation");
-    pyclass.def("kernel_map", &svKernelMap<StateVectorT>,
-                "Get internal kernels for operations");
+
+    pyclass
+        .def(py::init([](std::size_t num_qubits) {
+            return new StateVectorT(num_qubits);
+        }))
+        .def("resetStateVector", &StateVectorT::resetStateVector)
+        .def(
+            "setBasisState",
+            [](StateVectorT &sv, const size_t index) {
+                sv.setBasisState(index);
+            },
+            "Create Basis State.")
+        .def(
+            "setStateVector",
+            [](StateVectorT &sv, const std::vector<std::size_t> &indices,
+               const np_arr_c &state) {
+                const auto buffer = state.request();
+                std::vector<ComplexT> state_in;
+                if (buffer.size) {
+                    const auto ptr = static_cast<const ComplexT *>(buffer.ptr);
+                    state_in = std::vector<ComplexT>{ptr, ptr + buffer.size};
+                }
+                sv.setStateVector(indices, state_in);
+            },
+            "Set State Vector with values and their corresponding indices")
+        .def(
+            "getState",
+            [](const StateVectorT &sv, np_arr_c &state) {
+                py::buffer_info numpyArrayInfo = state.request();
+                auto *data_ptr =
+                    static_cast<std::complex<PrecisionT> *>(numpyArrayInfo.ptr);
+                if (state.size()) {
+                    std::copy(sv.getData(), sv.getData() + sv.getLength(),
+                              data_ptr);
+                }
+            },
+            "Copy StateVector data into a Numpy array.")
+        .def(
+            "UpdateData",
```

Comment:
Does this function name match the others for LGPU and LK too?

Reply:
No, but this was on purpose. The name we use for LGPU and LKokkos is `DeviceToHost`, this name makes sense for LGPU, does not always make sense for LKokkos, and never makes sense for LQubit.